sql/lantern.sql

```diff
@@ -29,14 +45,11 @@ BEGIN
 	END IF;
 
 	IF pgvector_exists THEN
-		-- taken from pgvector so our index can work with pgvector types
-		CREATE FUNCTION vector_l2sq_dist(vector, vector) RETURNS float8
-			AS 'MODULE_PATHNAME' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
-
+                -- Define operator class for vector type
 		CREATE OPERATOR CLASS dist_vec_l2sq_ops
 			DEFAULT FOR TYPE vector USING lantern_hnsw AS
 			OPERATOR 1 <-> (vector, vector) FOR ORDER BY float_ops,
-			FUNCTION 1 vector_l2sq_dist(vector, vector);
+			FUNCTION 1 l2sq_dist(real[], real[]);
```

Comment:
I think this works because now every time the operator is invoked, the inputs of type `vector` are implicitly cast to type `real[]`. Is this the case?
If so, how expensive is the cast? Can you point to the pgvector code implementing the cast? Can we get away without the cast by just defining a C function that understands the `vector` type and directly extracts floats from there for distance comparison?

context: 
- type casts https://www.postgresql.org/docs/current/sql-createcast.html
- custom types https://www.postgresql.org/docs/current/sql-createtype.html